errors.go

```diff
@@ -152,6 +152,12 @@ func GetUserMessage(err error) string {
 	return ""
 }
 
+
+// SetUserMessage sets a user readable error message.
+func (c *customError) SetUserMessage(err error, msg string) {
```

Comment:
This `err` is not used. Can we remove it?

Reply:
You should actually use ErrorType as a receiver.
Look at `Set` - it's a good example.